Integrations/src/main/java/io/deephaven/integrations/learn/Input.java

```diff
@@ -0,0 +1,45 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.v2.sources.ColumnSource;
+import org.jpy.PyObject;
+
+/**
+ * Provides an interface for getting data directly out of a table and into a function.
+ */
+public class Input {
+
+    private final Table table;
+    private final String[] colNames;
+    private final PyObject func;
+
+    /**
+     * Creates a new Input.
+     *
+     * @param table Table that contains data to be used with this function.
+     * @param colNames Array of column names to be extracted from this table.
+     * @param func Function that determines how data from this table is collected.
+     */
+    public Input(Table table, String[] colNames, PyObject func) {
+
+        this.table = table;
+        this.colNames = colNames;
+        this.func = func;
+    }
+
+    /** Creates an array of column sources specified by this table and given column names.
+     *
+     * @return Column sources with these column names from this table.
+     */
+    public ColumnSource<?>[] createColumnSource() {
```

Comment:
1. Done.
2. No.
3. (with all the relevant qualifiers pertaining to my inexperience) This seems to contradict the ethos of keeping pieces small and orthogonal. By `func` I presume you mean the gather function. If so, doesn't performing that computation here violate your above comment that "Input should just be a specification or recipe for how to gather data into an input to a later function call"?
If I'm wrong about this apparent discrepancy, why?
If not, I suspect you still know what you're talking about, so what is it about this particular problem that makes this apparent discrepancy permissible?

I don't mind moving it and it seems that you don't feel strongly either way, I'm just curious what the logic here is because I can't see it.